### ISSUE TITLE
Search results are all for non-round pebble (round app exists)

### ISSUE THREAD

**Issue by JVimes** [pebble-dev/rebble-store#143]
If I use the app store search feature (in the Pebble Android app), then try to install a resulting app, I get a pop-up saying "Sorry! This app is not compatible with your Pebble smartwatch" and it shows pictures of square screen watches (I have Time Round). However, if I manually browse to the same app (e.g. Timer, or Timer+) not using the search feature, then it installs fine.

Maybe the search feature is returning the wrong edition of the results for Time Round watches?

**Comment by mhelmetag** [comment]
Just for help debugging "chalk" is the hardware name for the pebble round.

**Comment by mhelmetag** [comment]
So... there is a settings page where you can specify your pebble version (this is only available in dev mode though). Setting the pebble version here will cause the search results to be correct.

Is the bug that this setting just isn't carried over from the app correctly?

I don't see any other way this could be set other than from the settings page so I'm not really convinced this is a bug vs just missing functionality.

**Comment by Willow-Systems** [comment]
For reference: the OG store sends the platform during the POST to agolia. Here is the search request from the beta store:

![image](https://user-images.githubusercontent.com/24474651/95562478-bb67eb80-0a0b-11eb-8261-0dd8a5dbb515.png)

and from the OG store:

![image](https://user-images.githubusercontent.com/24474651/95562499-c3279000-0a0b-11eb-8346-b57aa0ca0c50.png)

Note the extra field in the OG store request `analyticsTags`, this is what tells the search engine what platform the user is using.

**Comment by sGerli** [comment]
Thats not the issue, you can see in here https://github.com/pebble-dev/rebble-store/blob/b4d5ecf61521b5558578deb2e946b1adb94b31ad/src/components/pages/Search.vue#L72 that the filter is getting set

![image](https://user-images.githubusercontent.com/3890044/95622486-d6008b80-0a30-11eb-8b37-b30c880924bb.png)

And the results are Chalk apps http://store-beta.rebble.io/faces/search?hardware=chalk

![image](https://user-images.githubusercontent.com/3890044/95622507-e153b700-0a30-11eb-932d-fed4bc7b63d9.png)

Probably more testing would be useful to see if there could be something else causing issues, but atleast on the current version it seems to be working.

**Comment by mhelmetag** [comment]
Right, the search works as long as the localstore has the right hardware type set but... the hardware type doesn't get set by the Pebble app and it seems like the only way to set it is through the settings page.

Just trying to figure out the real issue and it doesn't seem like a "bug". More like missing functionality in that the Pebble app passing Pebble data to the appstore on index so that localstore has the right info.

**Comment by sGerli** [comment]
I'm not sure about Android but atleast on iOS with the updated boot settings it does get set by the Pebble app.

```json
{
    "config": {
        "webviews": {
            "appstore/search": "https://store-beta.rebble.io/faces/search?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/watchapps": "https://store-beta.rebble.io/apps?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/watchfaces": "https://store-beta.rebble.io/faces?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/application": "https://store-beta.rebble.io/app/$$id$$?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/search/query": "https://store-beta.rebble.io/faces/$$search_type$$/search?accessToken={access_token}&native=true&query=$$query$$&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/developer_apps": "https://store-beta.rebble.io/author/$$id$$?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$",
            "appstore/application_share": "https://store-beta.rebble.io/app/$$id$$",
            "appstore/application_changelog": "https://store-beta.rebble.io/app/$$id$$/versions?&accessToken={access_token}&platform={platform}&releaseId=207&appVersion=4.4&pebbleColor=$$pebbleColor$$&hardware=$$hardware$$&jsv=200&uid=$$user_id$$&mid=$$phone_id$$&pid=$$pebble_id$$&inApp=true&$$extras$$"
        }
    }
}
```

There are some url parameters that should be getting replaced by the app.
